.circleci/config.yml

```diff
@@ -0,0 +1,242 @@
+    description: "Install dependencies and build"
+    parameters:
+      working_directory:
+        default: .
+        type: string
+    steps:  
+      - run:
+          name: Install Dependencies
+          command: flutter pub get
+      - run:
+          name: Build
+          command: flutter packages pub run build_runner build
+          
+  run-flutter-integration-tests:
+    description: "Run Flutter integration tests"
+    steps:
+      - run:
+          name: Run tests
+          command: |
+            cd example
+            flutter drive --dart-define=API_KEY=$API_KEY --target=integration_test/app_test.dart \
+            --driver=test_driver/integration_test.dart
+
+  install-and-build-flutter:
+    description: "Install and build Flutter"
+    steps:
+      - run:
+          name: Install Flutter Dependencies
+          command: |
+            cd example
+            flutter pub get
+      - run:
+          name: Build
+          command: |
+            cd example
+            flutter packages pub run build_runner build
+
+  simulator-dependencies:
+    description: "Install iOS simulator dependencies"
+    steps:
+      - run:
+          name: "Install applesimutils"
+          command: |
+            HOMEBREW_NO_AUTO_UPDATE=1 brew tap wix/brew
+            HOMEBREW_NO_AUTO_UPDATE=1 brew install applesimutils
+
+  gradle-dependencies:
+    description: "Get Gradle dependencies"
+    steps:
+      - run:
+          working_directory: ./android
+          name: Chmod permissions
+          command: sudo chmod +x ./gradlew
+      - restore_cache:
+          name: Restore Gradle cache
+          key: v1-gradle-{{ checksum "./android/build.gradle" }}-{{ checksum  "./android/app/build.gradle" }}-{{ arch }}
+      - run:
+          working_directory: ./android
+          name: Download Gradle dependencies
+          command: ./gradlew dependencies
+      - save_cache:
+          name: Save Gradle cache
+          paths:
+            - ~/.gradle
+          key: v1-gradle-{{ checksum "./android/build.gradle" }}-{{ checksum  "./android/app/build.gradle" }}-{{ arch }}
+
+  android-sdk-dependencies:
+    description: "Install and set android SDK"
+    steps:
+      - run:
+          name: set ANDROID_SDK_ROOT
+          command: |
+            echo 'export ANDROID_SDK_ROOT=$HOME/android-tools'  >> $BASH_ENV
+      - restore_cache:
+          key: android=tools-v1-{{ checksum "scripts/install-android-tools.sh" }}-{{ arch }}
+
+      - run:
+          name: install android tools
+          command: |
+            sh scripts/install-android-tools.sh
+            echo 'export PATH=$ANDROID_SDK_ROOT/tools/bin:$PATH'  >> $BASH_ENV
+            echo 'export PATH=$ANDROID_SDK_ROOT/tools:$PATH'  >> $BASH_ENV
+            echo 'export PATH=$ANDROID_SDK_ROOT/platform-tools:$PATH'  >> $BASH_ENV
+            echo 'export PATH=$ANDROID_SDK_ROOT/emulator:$PATH'  >> $BASH_ENV
+            source $BASH_ENV
+            sdkmanager --list
```

Comment:
does `sdkmanager --list` have a side-effect or is it for debug purposes?

Reply:
I think I found it somewhere else. I believe it's just for debugging purposes since it will display all the installed images. Do you think we should remove it?

Reply:
I don't mind it, but let's maybe add a comment